File: python/writer/generators/test_circle.py
# SPDX-License-Identifier: MIT
import math
from pathlib import Path

from .common import run


def generate(skppy, destination: Path) -> None:
    model = skppy.new_model()
    model.entities.add_arc_curve((0, 0, 0), (0, 0, 1), 10.0, 0.0, 2.0 * math.pi, 24)
    skppy.save(model, destination)


if __name__ == "__main__":
    raise SystemExit(run(generate, "circle"))

File: c/validators/writer/test_circle.c
// SPDX-License-Identifier: MIT
#include "test_utils.h"

static bool validate_circle(const char* directory) {
    return validate_single_arc(
        "circle", directory, "circle.skp", 24, 10.0, 6.283185307179586, true);
}

WRITER_VALIDATOR_MAIN("circle", validate_circle)
